friends/src/components/UpdateFriendForm.js

```diff
@@ -0,0 +1,49 @@
+import React, {Component} from "react";
+import { connect } from 'react-redux';
+
+import { updateFriend } from '../actions';
+
+class UpdateFriendForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      age: '',
+      email: '',
+    }
+  }
+
+  componentDidMount() {
+    const { friends, match } = this.props;
+    const friend = friends.find(item => item.id === Number(match.params.id))
```

Comment:
Have you experimented with other ways of finding the specific friend object you're looking for?